src/schematools/validation.py

```diff
@@ -127,6 +127,28 @@ def run_all(self) -> Iterator[ValidationError]:
             yield from validator_inst.validate()
 
 
+class CamelCaseValidator(Validator):
+    """Checks that conversion to snake case and back leaves field identifiers unchanged."""
+
+    def validate(self) -> Iterator[ValidationError]:
+        """Run validation."""
+        for table in self.dataset.tables:
+            for field in table.fields:
+                error = _validate_camelcase(field.id)
+                if error is not None:
+                    yield error
+
+
+def _validate_camelcase(ident: str) -> Iterator[ValidationError]:
+    if ident == "":
+        return ValidationError("CamelCaseValidator", "empty identifier not allowed")
+    camel = toCamelCase(to_snake_case(ident))
+    if camel == ident:
+        return
+    msg = f"{ident} does not survive conversion to snake case and back, suggestion: {camel}"
+    return ValidationError("CamelCaseValidator", msg)
+
+
```

Comment:
There are some MyPy errors:

```
schematools/validation.py:139: error: Incompatible types in "yield" (actual type "Iterator[ValidationError]", expected type "ValidationError")
schematools/validation.py:144: error: Incompatible return value type (got "ValidationError", expected "Iterator[ValidationError]")
schematools/validation.py:147: error: Return value expected
schematools/validation.py:149: error: Incompatible return value type (got "ValidationError", expected "Iterator[ValidationError]")
```

Reply:
Easily resolved by changing the function return type on line 142 to `Optional[ValidationError]` and the return statement on line 147 to `return None`.